### PR TITLE
Add APFS Support

### PR DESCRIPTION
**Pull request by dfjxs**
Adding support for APFS via pre-processors.

APFS volumes can't be "attached", so we attach the containing partition, or the whole disk if there isn't a containing partition.

### CODE REVIEW

File: turbinia/processors/mount_local.py [hunk 1]
```diff
@@ -83,6 +83,86 @@ def GetDiskSize(source_path):
...
+        mount_cmd.extend(['-p', credential_data])
+      elif credential_type == 'recovery_password':
+        mount_cmd.extend(['-r', credential_data])
+      else:
+        # Unsupported credential type, try the next
+        log.warning(
+            'Unsupported credential type: {0!s}'.format(credential_type))
+        continue
+      mount_cmd.extend(['-X', 'allow_other', source_path, mount_path])
+      # Not logging command since it will contain credentials
```
Comment (aarontp) [inline turbinia/processors/mount_local.py:138]: Maybe we could still have a log message here, but without the full command line?  Something like "Using fsapfsmount with credential type {...} to mount {foo} on {bar}"?

File: turbinia/processors/mount_local.py [hunk 2]
```diff
@@ -414,8 +496,18 @@ def PreprocessMountPartition(partition_path, filesystem_type):
...
     subprocess.check_call(mount_cmd)
   except subprocess.CalledProcessError as exception:
+    if filesystem_type == 'EXT':
+      # ext2 will not mount with the noload option, so this may be the cause of
+      # the error.
+      mount_cmd = ['sudo', 'mount', '-o', 'ro', partition_path, mount_path]
+      log.info('Mount failed, trying: {0:s}'.format(' '.join(mount_cmd)))
+      try:
+        subprocess.check_call(mount_cmd)
+      except subprocess.CalledProcessError as exception:
```
Comment (aarontp) [inline turbinia/processors/mount_local.py:506]: Is this safe to do within the exception handler for the same exception type?

**Review by aarontp — APPROVED** [review]
LGTM, just a couple very minor comments/questions.

File: turbinia/processors/mount_local.py (hunk 1 above)
Comment (dfjxs) [reply turbinia/processors/mount_local.py:138]: Sure, I also added a similar log message for bdemount.

File: turbinia/processors/mount_local.py (hunk 2 above)
Comment (dfjxs) [reply turbinia/processors/mount_local.py:506]: It worked fine through all my testing, but it isn't very readable.
I've updated it.

**Comment by aarontp** [comment]
LGTM

**Pull request merged**